run_tests.py.in

```diff
@@ -593,27 +560,25 @@ def main():
     clean_test_tree(test_blddir)
     tests = filter_tests(test_blddir)
 
+    results = []
     with multiprocessing.Pool(
             processes=jobs,
             initializer=init_process,
             # Note: the arguments order is important here
             initargs=(test_blddir, args.skeleton, args.keep_temp_files,
                       args.valgrind, args.wine,)
     ) as pool:
-        pool.map(run_pack, chunks(tests, int(len(tests) / jobs + 1)))
+        results = pool.map(run_test, tests)
```

Comment:
As a tiny micro-optimization you can also switch to `imap` from `map`: https://stackoverflow.com/questions/26520781/multiprocessing-pool-whats-the-difference-between-map-async-and-imap.

As I understand it will allow for slightly faster output of first element. And will also at some later point turning `tests` itself into an iterable instead of an in-memory list.

Reply:
This small change took a few hours of debug. :D

First, it was necessary to change the way we get results: instead of `results = pool.map(...)` to have `for result in pool.imap(...)`, otherwise the script hangs on some futex and doesn't start running the tests:
```
$ time ./run_tests.py 
Running in 6 job(s)

```
strace looks like this:
```
strace -f -p $(ps a | grep run_tests | grep -v grep | egrep -o "^[0-9]+")
strace: Process 26953 attached
futex(0x558aaac2a270, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 0, NULL, FUTEX_BITSET_MATCH_ANY
```

Second, after I managed to get the tests running I started to have another problem: once in a few runs tet runner failed because an intermediate file got deleted from under a running thread. It turns out there was a race caused by two tests generating header with the same name. When I changed `map()` to `imap()` or even better `imap_unordered()` the two tests started to run in such a way that one of them removed the header while the other one was still processing it. One of the failed tests was a trivial modification of the other. It goes back to ancient times when it was useful, but not anymore, so I just removed the other test. Now I run test runner in a loop and it doesn't fail.

But it is a bad thing that intermediate files produced by different tests can collide in principle.

So, the end result is that run_tests.py use `imap_unordered`.